ui: Add column selector and filter to sessions page

gtr opened the pull request:
Closes [#73463](https://cockroachlabs.atlassian.net/browse/CRDB-11600)

Previously, the sessions page was missing the column selector and filter
components that the stmts and txns page have. This change adds both
components in addition to new columns in the sessions overview page and
edits to the sessions details page.



Release note (ui change): added column selector, filters, new columns to
the sessions page and sessions details.

Screenshots:

Sessions overview page:
![sessions-overview-page](https://user-images.githubusercontent.com/35943354/153674473-639b23a5-031a-4d33-90cc-87db24ec7c4a.png)


Sessions overview page with column selector:
![sessions-overview-page-column-selector](https://user-images.githubusercontent.com/35943354/153674485-9503434f-0860-4b84-9ad4-0175586bd986.png)


Sessions overview page with results from column selector:
![sessions-overview-page-column-selector-results](https://user-images.githubusercontent.com/35943354/153674490-788627c7-0392-4d15-a2e5-9c2e79de92f9.png)


Sessions overview page with filter:
![sessions-overview-page-filter](https://user-images.githubusercontent.com/35943354/153674501-a83394ac-2a8a-4035-a030-4f2b2f402ab2.png)


Sessions overview page with results from filter:
![sessions-overview-page-filter-results](https://user-images.githubusercontent.com/35943354/153674508-5b98f52d-7ea0-48f9-bef4-88635878de71.png)


Sessions detail page:
![Screen Shot 2022-02-11 at 4 33 14 PM](https://user-images.githubusercontent.com/35943354/153673353-b984462e-092c-4849-b03a-9a33091e4f45.png)

Commit:
ui: Add column selector and filter including session details to the session overview page

Closes [#73463](https://cockroachlabs.atlassian.net/browse/CRDB-11600)
Previously, the sessions page was missing the column selector and filter
components that the stmts and txns page have. This change adds both
components in addition to new columns in the sessions overview page and
edits to the sessions details page.

Release note (ui change): added column selector, filters, new columns to
the sessions page and sessions details.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75965)
<!-- Reviewable:end -->

gtr commented:
@kevin-v-ngo A few things in the sessions details page can't be implemented until we fix [#67888](https://github.com/cockroachdb/cockroach/issues/67888). This includes details for the "most recent transaction" and "most recent statement" sections since we don't currently store information about non-active queries.

ericharmeling reviewed (commented):
> Reviewed 1 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @kevin-v-ngo, @maryliag, and @xinhaoz)

___
*[pkg/ui/workspaces/cluster-ui/src/icon/circleFilled.tsx, line 21 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv-ouNC-yL90p5d7tZD:-Mv-ouND89jHzKWJog7g:b95fitw) ([raw file](https://github.com/cockroachdb/cockroach/blob/a21e18227b22a72ac04a4260e8a1d9ea65b801ac/pkg/ui/workspaces/cluster-ui/src/icon/circleFilled.tsx#L21)):*
> ```XML
>   const { fill, className } = props;
>   return (
>     <svg height={10} width={20} className={className} {...props}>
> ```

I think the height and width and fill can all be specified in the class definition (e.g., `sessionPage.module.scss` file) and then passed to the component via the `className`.

I'm doing it like this in a `.module.scss` file:

```
.bool-setting-icon {

  &__enabled {
    fill: $colors--success;
    margin-right: 8px;
    height: 8px;
    width: 8px;
  }

  &__disabled {
    fill: $colors--disabled;
    margin-right: 8px;
    height: 8px;
    width: 8px;
  }
}
```

and then in the parent component:
```
<CircleFilled className={cx("bool-setting-icon__enabled")} />
```


<!-- Sent from Reviewable.io -->

kevin-v-ngo commented:
> @kevin-v-ngo A few things in the sessions details page can't be implemented until we fix [#67888](https://github.com/cockroachdb/cockroach/issues/67888). This includes details for the "most recent transaction" and "most recent statement" sections since we don't currently store information about non-active queries.

Thanks. What's the "last_active_query" column in crdb_internal.cluster_sessions?

maryliag reviewed (changes requested):
From the screenshot both column selector and filter are not in the right location, with the correct spacing, etc

> Reviewed 1 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-Mv05eXR8897FACRIGmY:b-ot41y7)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @kevin-v-ngo, and @xinhaoz)

___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 328 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv06pVWAnr1ElGX53-i:-Mv06pVWAnr1ElGX53-j:b-cji6r) ([raw file](https://github.com/cockroachdb/cockroach/blob/a21e18227b22a72ac04a4260e8a1d9ea65b801ac/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L328)):*
> ```XML
> 
>     const appNames = getSessionAppFilterOptions(sessionsData);
>     const nodes = isTenant
> ```

we are not really filtering anything based on nodes, so all mentions of nodes and tenants can be removed
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTableContent.tsx, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv07CYdCDMtMsWZhzuN:-Mv07CYe4mHQc2gXexM7:b-5ynxe0) ([raw file](https://github.com/cockroachdb/cockroach/blob/a21e18227b22a72ac04a4260e8a1d9ea65b801ac/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTableContent.tsx#L26)):*
> ```XML
>       content={"The timestamp at which the session started."}
>     >
>       Session Start Time (UTC)
> ```

since you created the label on `sessionsColumnLabels` you can use that here, so is one source of truth



_Code quote (from pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx):_
```XML
sessionsColumnLabels
```


<!-- Sent from Reviewable.io -->

kevin-v-ngo reviewed (commented):
What's the "last_active_query" column in crdb_internal.cluster_sessions? I thought that was the most recent query which we can add to the UI. 

From the screenshot, the spacing seems a bit off. Adding @Annebirzin for another set of eyes.

gtr commented:
> From the screenshot both column selector and filter are not in the right location, with the correct spacing, etc
> 
> > Reviewed 1 of 11 files at r1.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-Mv05eXR8897FACRIGmY:b-ot41y7)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @kevin-v-ngo, and @xinhaoz)
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 328 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv06pVWAnr1ElGX53-i:-Mv06pVWAnr1ElGX53-j:b-cji6r) ([raw file](https://github.com/cockroachdb/cockroach/blob/a21e18227b22a72ac04a4260e8a1d9ea65b801ac/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L328)):_
> 
> > ```
> >     const appNames = getSessionAppFilterOptions(sessionsData);
> >     const nodes = isTenant
> > ```
> 
> we are not really filtering anything based on nodes, so all mentions of nodes and tenants can be removed
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTableContent.tsx, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv07CYdCDMtMsWZhzuN:-Mv07CYe4mHQc2gXexM7:b-5ynxe0) ([raw file](https://github.com/cockroachdb/cockroach/blob/a21e18227b22a72ac04a4260e8a1d9ea65b801ac/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTableContent.tsx#L26)):_
> 
> > ```
> >       content={"The timestamp at which the session started."}
> >     >
> >       Session Start Time (UTC)
> > ```
> 
> since you created the label on `sessionsColumnLabels` you can use that here, so is one source of truth
> 
> _Code quote (from pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx):_
> 
> ```
> sessionsColumnLabels
> ```

done, good catches! 👍🏽

gtr commented:
> What's the "last_active_query" column in crdb_internal.cluster_sessions? I thought that was the most recent query which we can add to the UI.
> 
> From the screenshot, the spacing seems a bit off. Adding @Annebirzin for another set of eyes.

It is! But it only actually stores the string of the query, not any other information such as the start time.

maryliag reviewed (commented):
From the images looks like your filter count is showing how many results there are and not how many filters where selected. The value should be how many filters where selected

> Reviewed 1 of 11 files at r1, 2 of 9 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-Mv5AtwU1W4kunQngLA7:b-23vfcn)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)

___
*[-- commits, line 8 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv5ADvL8LJKTe0vgA7Q):*
nit: add which issue this is fixing


<!-- Sent from Reviewable.io -->

Annebirzin commented:
Thanks @gtr! A couple of notes and questions:

Session Overview
-  For the Filter button, the text looks off-center inside the button, and the arrow icon looks very small. Can we match the alignment/styling of other Filter buttons on Statements and Transactions?
- The `1-20 of 35 results` pagination label looks vertically misaligned. Can we center align this vertically with the column selector button?

Session Detail 
- Update the `Statement` header above the Statement SQL box to `Most Recent Statement`
- Fix the spacing of the 'View Statement Details' link at the bottom of the page? (add padding above the link) 
- Use our core blue color for links on this page: Info-blue `#0055FF` (Applies to `Gateway node` and `View statement detail` links) 
- @kevin-v-ngo should we also update the `Transaction` header to `Most Recent Transaction` and add the Transaction SQL box? Or is that for a future update?

kevin-v-ngo commented:
> Thanks @gtr! A couple of notes and questions:
> 
> Session Overview
> 
> * For the Filter button, the text looks off-center inside the button, and the arrow icon looks very small. Can we match the alignment/styling of other Filter buttons on Statements and Transactions?
> * The `1-20 of 35 results` pagination label looks vertically misaligned. Can we center align this vertically with the column selector button?
> 
> Session Detail
> 
> * Update the `Statement` header above the Statement SQL box to `Most Recent Statement`
> * Fix the spacing of the 'View Statement Details' link at the bottom of the page (add padding above the link)
> * Use our core blue color for links on this page: Info-blue `#0055FF` (Applies to `Gateway node` and `View statement detail` links)
> * @kevin-v-ngo should we also update the `Transaction` header to `Most Recent Transaction` and add the Transaction SQL box? Or is that for a future update?

It should be consistent with statement - "Most Recent Transaction". @gtr, any reason why we don't have the SQL for the most recent transaction? For **active** Sessions, we should have it correct? I'm not sure for **idle** Sessions.

gtr commented:
> > Thanks @gtr! A couple of notes and questions:
> > Session Overview
> > 
> > * For the Filter button, the text looks off-center inside the button, and the arrow icon looks very small. Can we match the alignment/styling of other Filter buttons on Statements and Transactions?
> > * The `1-20 of 35 results` pagination label looks vertically misaligned. Can we center align this vertically with the column selector button?
> > 
> > Session Detail
> > 
> > * Update the `Statement` header above the Statement SQL box to `Most Recent Statement`
> > * Fix the spacing of the 'View Statement Details' link at the bottom of the page (add padding above the link)
> > * Use our core blue color for links on this page: Info-blue `#0055FF` (Applies to `Gateway node` and `View statement detail` links)
> > * @kevin-v-ngo should we also update the `Transaction` header to `Most Recent Transaction` and add the Transaction SQL box? Or is that for a future update?
> 
> It should be consistent with statement - "Most Recent Transaction". @gtr, any reason why we don't have the SQL for the most recent transaction? For **active** Sessions, we should have it correct? I'm not sure for **idle** Sessions.

The way the sessions details page is currently written, the `cockroach.server.serverpb.ITxnInfo` transaction type doesn't contain the SQL text but it should be available for active sessions correct. I couldn't get it to work for this current change but I'm working on it right now.

gtr commented:
> Thanks @gtr! A couple of notes and questions:
> 
> Session Overview
> 
> * For the Filter button, the text looks off-center inside the button, and the arrow icon looks very small. Can we match the alignment/styling of other Filter buttons on Statements and Transactions?
> * The `1-20 of 35 results` pagination label looks vertically misaligned. Can we center align this vertically with the column selector button?
> 
> Session Detail
> 
> * Update the `Statement` header above the Statement SQL box to `Most Recent Statement`
> * Fix the spacing of the 'View Statement Details' link at the bottom of the page (add padding above the link)
> * Use our core blue color for links on this page: Info-blue `#0055FF` (Applies to `Gateway node` and `View statement detail` links)
> * @kevin-v-ngo should we also update the `Transaction` header to `Most Recent Transaction` and add the Transaction SQL box? Or is that for a future update?

Thank you! Done 👍🏽

kevin-v-ngo commented:
Hi @gtr , i've added to the following [task](https://github.com/cockroachdb/cockroach/issues/74257) to track the missing "Most recent transaction" information.

maryliag reviewed (commented):
this is looking great! 
One thing: the `session start time` and `actions` columns should not be "selectable", meaning, they should always be displayed. We have a flag for alwaysShow that you can use on that column (e.g. https://github.com/cockroachdb/cockroach/blob/34666243f647931d5ee65ed9d9b6ee294647cdf5/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L259 )

> Reviewed 2 of 9 files at r2, 1 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvOrXL7AD3-sC6Ma8bp:b1jvrga)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)

___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 233 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOt-7_AiaoZ5Qiu5el:-MvOt-7_AiaoZ5Qiu5em:b-s7h6vs) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L233)):*
> ```XML
>         timeUnit: filters.timeUnit,
>         regions: filters.regions,
>         nodes: filters.nodes,
> ```

nit: you don't need regions and nodes here
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 256 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOt2wc63HwQZ4jl0EN:-MvOt2wc63HwQZ4jl0EO:bsc9lul) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L256)):*
> ```XML
>         timeUnit: undefined,
>         regions: undefined,
>         nodes: undefined,
> ```

you don't need regions and nodes here
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 275 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOtxgX7S5XqCdlTkzZ:-MvOtxgX7S5XqCdlTkz_:b-t7nk1q) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L275)):*
> ```XML
>     }
>     const activeFilters = calculateActiveFilters(filters);
>     const internalAppNamePrefix = "$ internal";
> ```

this value should come from the props, we don't want to set here, in case it changes
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.tsx, line 124 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOuGpr5Zpys_BmD5mc:-MvOuGpr5Zpys_BmD5md:bioh7yw) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.tsx#L124)):*
> ```XML
>           action: "Terminate Statement",
>         }),
>       onFilterChange: (value: Filters) => {
> ```

make sure you're also testing this on CC Console
___
*[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 266 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOuqzE6TT5OJuL7-qD:-MvOuqzE6TT5OJuL7-qE:b25x25g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L266)):*
> ```XML
>         }
>       >
>         Statements Run
> ```

nit: use the getLabel here too
___
*[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 277 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOutGwCpaHbInGmg9F:-MvOutGwCpaHbInGmg9G:b25x25g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L277)):*
> ```XML
>         content={"The duration of the open transaction, if there is one."}
>       >
>         Transaction Duration
> ```

nit: use the getLabel here too
___
*[pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx, line 30 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOv17kEd5m24R5CIU6:-MvOv17kEd5m24R5CIU7:b-pxv276) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx#L30)):*
> ```XML
>   terminateSessionAction,
> } from "src/redux/sessions/sessionsSagas";
> import { nodeRegionsByIDSelector } from "oss/src/redux/nodes";
> ```

nit: you don't need nodes on this page
___
*[pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx, line 73 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOv3Uq6seyDmFowOQq:-MvOv3Uq6seyDmFowOQr:b-pxv276) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx#L73)):*
> ```XML
>       columns: sessionColumnsLocalSetting.selectorToArray(state),
>       filters: filtersLocalSetting.selector(state),
>       nodeRegions: nodeRegionsByIDSelector(state),
> ```

nit: you don't need nodes on this page
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.stories.tsx, line 28 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvHMh1moyPMdPRECy:-MvOvHMh1moyPMdPRECz:b-p5ixmb) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bd7ec036fffcf2e3b07df7d214aa2a700ffe7ba/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.stories.tsx#L28)):*
> ```XML
>   Store,
> } from "redux";
> import { SessionsPageConnected } from "./sessionsPageConnected";
> ```

why this removal? Don't we want to test this page?
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 20 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvQjR4NUH7qmhgSL7:-MvOvQjS-Cxq_NAz9SyC:b-c13oi3) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L20)):*
> ```SCSS
> 
> .sessions-filter {
>   font-size: 14px;
> ```

use the variable `$font-size--medium` instead of the direct value here
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 21 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvxyg2n8OozKWbm0q:-MvOvxyg2n8OozKWbm0r:b-t85rae) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L21)):*
> ```SCSS
> .sessions-filter {
>   font-size: 14px;
>   margin-bottom: 10px;
> ```

use `$spacing-smaller` instead
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 25 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvzwd6r5G9UEdqEKd:-MvOvzwd6r5G9UEdqEKe:b-rhx3ew) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L25)):*
> ```SCSS
> 
> .session-column-selector {
>   margin-bottom: 10px;
> ```

use `$spacing-smaller`


<!-- Sent from Reviewable.io -->

jocrl reviewed (changes requested):
Yay, great stuff! Just a few nits.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvPaSNYAZdpNUCHM3Bi:b-okjvev)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)

___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 262 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPFeBYAdtEFlehBHIO:-MvPFeBYAdtEFlehBHIP:bs6ggbw) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L262)):*
> ```XML
>   };
> 
>   filteredSessionsData = (): {
> ```

nit: please verb-ify this and other function names that currently look like nouns. `s/filteredSessionsData/getFilteredSessionsData`
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 294 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPGBuG6lcJDUwy3dQu:-MvPGBuG6lcJDUwy3dQv:b-yij243) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L294)):*
> ```XML
>             (showInternal && isInternal(s)) ||
>             s.session.application_name === filters.app ||
>             apps.includes(s.session.application_name)
> ```

nit: Is it just me, or are these two lines redundant?
If the apps are getting string split above, let's just work with the split version.



_Suggestion:_
```XML
            s.session.application_name === filters.app ||
            apps.includes(s.session.application_name)
```
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 335 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPWxQv7CpW9wzIgE_n:-MvPWxQv7CpW9wzIgE_o:bw2m16g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L335)):*
> ```XML
>     const isColumnSelected = (c: ColumnDescriptor<SessionInfo>) => {
>       return (
>         ((userSelectedColumnsToShow === null ||
> ```

nit: perhaps simplify this chunk by generally checking if values are truthy?
```
(!userSelectedColumnsToShow && c.showByDefault) ||
(userSelectedColumnsToShow && userSelectedColumnsToShow.includes(c.name)) ||
c.alwaysShow
```
___
*[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx, line 147 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvP_PGc86THCX2baRna:-MvP_PGd1Ra3joR5OjF2:b-4hifk7) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx#L147)):*
> ```XML
>   onTerminateStatementClick?: () => void,
> ): ColumnDescriptor<SessionInfo>[] {
>   const statType = "session";
> ```

nit: an explicit type annotation here would be nice, to make it easier to understand what other possible values are
`s/statType/statType: StatisticType`
___
*[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 117 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvP_rXM0BvicpyvkdWv:-MvP_rXM0BvicpyvkdWw:b7vx7j4) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L117)):*
> ```XML
> // StatisticType is used to modify the content of the tooltip.
> export const statisticsTableTitles: StatisticTableTitleType = {
>   sessionStart: () => {
> ```

nit: Please verb-ify these functions. I'd also suggest naming them to indicate that they are returning components, e.g. `s/sessionStart/getSessionStartComponent`



_Code quote:_
```XML
sessionStart
```


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> Reviewed 1 of 9 files at r2, 1 of 4 files at r3, 7 of 9 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)

___
*[-- commits, line 8 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-Mv5ADvL8LJKTe0vgA7Q):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

nit: add which issue this is fixing
</blockquote></details>

you added to the PR description, but not here :) 


<!-- Sent from Reviewable.io -->

gtr commented:
> Yay, great stuff! Just a few nits.
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvPaSNYAZdpNUCHM3Bi:b-okjvev)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 262 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPFeBYAdtEFlehBHIO:-MvPFeBYAdtEFlehBHIP:bs6ggbw) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L262)):_
> 
> > ```
> >   };
> > 
> >   filteredSessionsData = (): {
> > ```
> 
> nit: please verb-ify this and other function names that currently look like nouns. `s/filteredSessionsData/getFilteredSessionsData`
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 294 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPGBuG6lcJDUwy3dQu:-MvPGBuG6lcJDUwy3dQv:b-yij243) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L294)):_
> 
> > ```
> >             (showInternal && isInternal(s)) ||
> >             s.session.application_name === filters.app ||
> >             apps.includes(s.session.application_name)
> > ```
> 
> nit: Is it just me, or are these two lines redundant? If the apps are getting string split above, let's just work with the split version.
> 
> _Suggestion:_
> 
> ```
>             s.session.application_name === filters.app ||
>             apps.includes(s.session.application_name)
> ```
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 335 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvPWxQv7CpW9wzIgE_n:-MvPWxQv7CpW9wzIgE_o:bw2m16g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L335)):_
> 
> > ```
> >     const isColumnSelected = (c: ColumnDescriptor<SessionInfo>) => {
> >       return (
> >         ((userSelectedColumnsToShow === null ||
> > ```
> 
> nit: perhaps simplify this chunk by generally checking if values are truthy?
> 
> ```
> (!userSelectedColumnsToShow && c.showByDefault) ||
> (userSelectedColumnsToShow && userSelectedColumnsToShow.includes(c.name)) ||
> c.alwaysShow
> ```
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx, line 147 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvP_PGc86THCX2baRna:-MvP_PGd1Ra3joR5OjF2:b-4hifk7) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsTable.tsx#L147)):_
> 
> > ```
> >   onTerminateStatementClick?: () => void,
> > ): ColumnDescriptor<SessionInfo>[] {
> >   const statType = "session";
> > ```
> 
> nit: an explicit type annotation here would be nice, to make it easier to understand what other possible values are `s/statType/statType: StatisticType`
> 
> _[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 117 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvP_rXM0BvicpyvkdWv:-MvP_rXM0BvicpyvkdWw:b7vx7j4) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L117)):_
> 
> > ```
> > // StatisticType is used to modify the content of the tooltip.
> > export const statisticsTableTitles: StatisticTableTitleType = {
> >   sessionStart: () => {
> > ```
> 
> nit: Please verb-ify these functions. I'd also suggest naming them to indicate that they are returning components, e.g. `s/sessionStart/getSessionStartComponent`
> 
> _Code quote:_
> 
> ```
> sessionStart
> ```

Done! :)

> this is looking great! One thing: the `session start time` and `actions` columns should not be "selectable", meaning, they should always be displayed. We have a flag for alwaysShow that you can use on that column (e.g.
> 
> https://github.com/cockroachdb/cockroach/blob/34666243f647931d5ee65ed9d9b6ee294647cdf5/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L259
> 
> )
> > Reviewed 2 of 9 files at r2, 1 of 4 files at r3.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvOrXL7AD3-sC6Ma8bp:b1jvrga)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 233 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOt-7_AiaoZ5Qiu5el:-MvOt-7_AiaoZ5Qiu5em:b-s7h6vs) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L233)):_
> 
> > ```
> >         timeUnit: filters.timeUnit,
> >         regions: filters.regions,
> >         nodes: filters.nodes,
> > ```
> 
> nit: you don't need regions and nodes here
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 256 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOt2wc63HwQZ4jl0EN:-MvOt2wc63HwQZ4jl0EO:bsc9lul) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L256)):_
> 
> > ```
> >         timeUnit: undefined,
> >         regions: undefined,
> >         nodes: undefined,
> > ```
> 
> you don't need regions and nodes here
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx, line 275 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOtxgX7S5XqCdlTkzZ:-MvOtxgX7S5XqCdlTkz_:b-t7nk1q) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPage.tsx#L275)):_
> 
> > ```
> >     }
> >     const activeFilters = calculateActiveFilters(filters);
> >     const internalAppNamePrefix = "$ internal";
> > ```
> 
> this value should come from the props, we don't want to set here, in case it changes
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.tsx, line 124 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOuGpr5Zpys_BmD5mc:-MvOuGpr5Zpys_BmD5md:bioh7yw) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.tsx#L124)):_
> 
> > ```
> >           action: "Terminate Statement",
> >         }),
> >       onFilterChange: (value: Filters) => {
> > ```
> 
> make sure you're also testing this on CC Console
> 
> _[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 266 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOuqzE6TT5OJuL7-qD:-MvOuqzE6TT5OJuL7-qE:b25x25g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L266)):_
> 
> > ```
> >         }
> >       >
> >         Statements Run
> > ```
> 
> nit: use the getLabel here too
> 
> _[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 277 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOutGwCpaHbInGmg9F:-MvOutGwCpaHbInGmg9G:b25x25g) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L277)):_
> 
> > ```
> >         content={"The duration of the open transaction, if there is one."}
> >       >
> >         Transaction Duration
> > ```
> 
> nit: use the getLabel here too
> 
> _[pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx, line 30 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOv17kEd5m24R5CIU6:-MvOv17kEd5m24R5CIU7:b-pxv276) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx#L30)):_
> 
> > ```
> >   terminateSessionAction,
> > } from "src/redux/sessions/sessionsSagas";
> > import { nodeRegionsByIDSelector } from "oss/src/redux/nodes";
> > ```
> 
> nit: you don't need nodes on this page
> 
> _[pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx, line 73 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOv3Uq6seyDmFowOQq:-MvOv3Uq6seyDmFowOQr:b-pxv276) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/db-console/src/views/sessions/sessionsPage.tsx#L73)):_
> 
> > ```
> >       columns: sessionColumnsLocalSetting.selectorToArray(state),
> >       filters: filtersLocalSetting.selector(state),
> >       nodeRegions: nodeRegionsByIDSelector(state),
> > ```
> 
> nit: you don't need nodes on this page
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.stories.tsx, line 28 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvHMh1moyPMdPRECy:-MvOvHMh1moyPMdPRECz:b-p5ixmb) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bd7ec036fffcf2e3b07df7d214aa2a700ffe7ba/pkg/ui/workspaces/cluster-ui/src/sessions/sessionsPageConnected.stories.tsx#L28)):_
> 
> > ```
> >   Store,
> > } from "redux";
> > import { SessionsPageConnected } from "./sessionsPageConnected";
> > ```
> 
> why this removal? Don't we want to test this page?
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 20 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvQjR4NUH7qmhgSL7:-MvOvQjS-Cxq_NAz9SyC:b-c13oi3) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L20)):_
> 
> > ```css-scss
> > .sessions-filter {
> >   font-size: 14px;
> > ```
> 
> use the variable `$font-size--medium` instead of the direct value here
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 21 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvxyg2n8OozKWbm0q:-MvOvxyg2n8OozKWbm0r:b-t85rae) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L21)):_
> 
> > ```css-scss
> > .sessions-filter {
> >   font-size: 14px;
> >   margin-bottom: 10px;
> > ```
> 
> use `$spacing-smaller` instead
> 
> _[pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss, line 25 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvOvzwd6r5G9UEdqEKd:-MvOvzwd6r5G9UEdqEKe:b-rhx3ew) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/sessions/sessionPage.module.scss#L25)):_
> 
> > ```css-scss
> > .session-column-selector {
> >   margin-bottom: 10px;
> > ```
> 
> use `$spacing-smaller`

Apologies for the delay; was waiting for [#76307](https://github.com/cockroachdb/cockroach/pull/76307) to merge. Done!

gtr commented:
Updated the screenshots to reflect:

- Updated column case
- New "Session Duration" label on filter
- Fixed width of 1200px for sessions details
- Updated font for "Most Recent Statement" row

jocrl reviewed (approved):
Thanks! Nits look good to me; I'll leave the stamp to Marylia 🙂

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvzSDhz8JsSYBYx_eml:bknou4x)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)

___
*[pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx, line 117 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-MvP_rXM0BvicpyvkdWv:-MvzSPNo2r6KS0G9EtB9:b-2g93il) ([raw file](https://github.com/cockroachdb/cockroach/blob/29071a787eb3efabf6083edaa1a65bf8e022ae52/pkg/ui/workspaces/cluster-ui/src/statsTableUtil/statsTableUtil.tsx#L117)):*
<details><summary><i>Previously, gtr (Gerardo Torres Castro) wrote…</i></summary><blockquote>

> > ```
> > // StatisticType is used to modify the content of the tooltip.
> > export const statisticsTableTitles: StatisticTableTitleType = {
> >   sessionStart: () => {
> > ```
> > 

> 
> nit: Please verb-ify these functions. I'd also suggest naming them to indicate that they are returning components, e.g. `s/sessionStart/getSessionStartComponent`
> 
> _Code quote:_
> 
> ```
> sessionStart
> ```
> 

Done! :)


> this is looking great! One thing: the `session start time` and `actions` columns should not be "selectable", meaning, they should always be displayed. We have a flag for alwaysShow that you can use on that column (e.g.
> 
> https://github.com/cockroachdb/cockroach/blob/34666243f647931d5ee65ed9d9b6ee294647cdf5/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L259
> 
> )
> 

> > Reviewed 2 of 9 files at r2, 1 of 4 files at r3.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvOrXL7AD3-sC6Ma8bp:b1jvrga)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)
> > 
</blockquote></details>

Not sure why the comment reply formatting is weird, but just following up that we chatted offline about not changing `sessionStart` because of the type definition on the key of `StatisticTableTitleType`


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
I want to make sure this is working on CC Console before I can give the approval :) 

> Reviewed 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MvzXtDv2p6pqs7c4ipV:b4nrxi7)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)



<!-- Sent from Reviewable.io -->

gtr commented:
Video of the sessions page tested on CC Console:

https://user-images.githubusercontent.com/35943354/154766490-510bc831-896a-427f-88f0-de2d221a1a19.mov

maryliag reviewed (approved):
Awesome! 
There is just one check that I would have like to see on the video: select a filter, change page, come back, the filter must be selected
(no need to create another video, just make sure that this behaviour works as expected :D ) 

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 11 files at r1, 2 of 2 files at r7, 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75965#-:-MwDrjggA7D4DWkKc2pU:bgfkr51)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @gtr, @maryliag, and @xinhaoz)



<!-- Sent from Reviewable.io -->

gtr commented:
bors r+

gtr commented:
bors retry
